compiler/src/iree/compiler/Dialect/Flow/Conversion/TensorToFlow/Utils.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_DIALECT_FLOW_CONVERSION_TENSORTOFLOW_UTILS_H_
+#define IREE_COMPILER_DIALECT_FLOW_CONVERSION_TENSORTOFLOW_UTILS_H_
+
+#include "llvm/ADT/SmallVector.h"
+
+namespace mlir {
+class Location;
+class LogicalResult;
+class OpBuilder;
+class RewriterBase;
+class Value;
+
+namespace tensor {
+class ExtractSliceOp;
+class InsertSliceOp;
+}  // namespace tensor
+
+namespace iree_compiler {
+namespace IREE {
+namespace Flow {
+
+/// Rewrite the given InsertSliceOp into a Flow::TensorUpdateOp.
+LogicalResult convertInsertSliceOpToFlowUpdateOp(
+    RewriterBase &rewriter, tensor::InsertSliceOp insertOp);
+
+/// Rewrite the given ExtractSliceOp into a Flow::TensorSliceOp.
+LogicalResult convertExtractSliceOpToFlowSliceOp(
+    RewriterBase &rewriter, tensor::ExtractSliceOp sliceOp);
+
+/// Generates `tensor.dim` operations to get the dynamic sizes of a value `v`.
+llvm::SmallVector<Value, 4> getDynamicDimValues(OpBuilder &b, Location loc,
```

Comment:
Recently added `tensor::getDimValues` that returns an `OpFoldResult`. Maybe use that in conjunction with `arith::getValueOrCreateConstantIndexOps`?